lib/common/client/base.rb

```diff
@@ -31,25 +31,26 @@ def perform(method, path, params, headers = nil)
 
       def request(method, path, params = {}, headers = {})
         raise_not_authenticated if headers.keys.include?('Token') && headers['Token'].nil?
+        binding.pry
         connection.send(method.to_sym, path, params) { |request| request.headers.update(headers) }.env
       rescue Faraday::ClientError, Timeout::Error
         raise Common::Client::Errors::ClientError
       end
 
-      def get(path, params, headers = base_headers)
-        request(:get, path, params, headers)
+      def get(path, params = nil, headers = nil)
```

Comment:
If I default them to {}, someone invoking the method could still pass in nil explicitly. I think if I build this as named arguments, that becomes less of an issue.

ie. If I have no params, but do have custom headers, I would currently need to do:

get('/path', {}, { custom_header_key: value })

But it's common to pass nil, in this scenario for params, hence why I'm doing `|| {}`

However if I have named arguments:

get(path: '/path', headers: { custom_header_key: key }) then there is no reason to pass the anything for params when there are none.